server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/event/ComputeMessagePreviewListener.java

```diff
@@ -0,0 +1,120 @@
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageIdManager;
+import org.apache.james.mailbox.events.Event;
+import org.apache.james.mailbox.events.Group;
+import org.apache.james.mailbox.events.MailboxListener;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.FetchGroup;
+import org.apache.james.mailbox.model.MessageId;
+import org.apache.james.mailbox.model.MessageResult;
+import org.apache.james.mailbox.store.SessionProvider;
+import org.apache.james.mime4j.dom.Message;
+import org.apache.james.mime4j.stream.MimeConfig;
+import org.apache.james.util.html.HtmlTextExtractor;
+import org.apache.james.util.mime.MessageContentExtractor;
+import org.apache.james.util.mime.MessageContentExtractor.MessageContent;
+
+import com.github.fge.lambdas.Throwing;
+import com.github.steveash.guavate.Guavate;
+
+import reactor.core.publisher.Flux;
+import reactor.core.scheduler.Schedulers;
+
+public class ComputeMessagePreviewListener implements MailboxListener.GroupMailboxListener {
+    public static class ComputeMessagePreviewListenerGroup extends Group {
+
+    }
+
+    private static final Group GROUP = new ComputeMessagePreviewListenerGroup();
+
+    private final MessageIdManager messageIdManager;
+    private final MessagePreviewStore messagePreviewStore;
+    private final SessionProvider sessionProvider;
+    private final MessageContentExtractor messageContentExtractor;
+    private final HtmlTextExtractor htmlTextExtractor;
+
+    @Inject
+    public ComputeMessagePreviewListener(SessionProvider sessionProvider, MessageIdManager messageIdManager,
+                                         MessagePreviewStore messagePreviewStore, MessageContentExtractor messageContentExtractor,
+                                         HtmlTextExtractor htmlExtractor) {
+        this.sessionProvider = sessionProvider;
+        this.messageIdManager = messageIdManager;
+        this.messagePreviewStore = messagePreviewStore;
+        this.messageContentExtractor = messageContentExtractor;
+        this.htmlTextExtractor = htmlExtractor;
+    }
+
+    @Override
+    public Group getDefaultGroup() {
+        return GROUP;
+    }
+
+    @Override
+    public void event(Event event) throws MailboxException {
+        if (event instanceof Added) {
+            MailboxSession session = sessionProvider.createSystemSession(event.getUsername());
+            handleAddedEvent((Added) event, session);
+        }
+    }
+
+    private void handleAddedEvent(Added addedEvent, MailboxSession session) throws MailboxException {
+        Flux.fromIterable(messageIdManager.getMessages(getEventMessageIds(addedEvent), FetchGroup.BODY_CONTENT, session))
+            .publishOn(Schedulers.boundedElastic())
+            .map(Throwing.function((MessageResult messageResult) -> Pair.of(messageResult.getMessageId(), computePreview(messageResult))).sneakyThrow())
```

Comment:
This sneakyThrow is not needed IMO (we don't care about the type of the exception)